src/modelbench/hazards.py

```diff
@@ -12,9 +12,14 @@
 from pydantic import BaseModel, ConfigDict
 
 from modelbench.scoring import BAND_CONSTANTS, LetterGradeMixin, NumericGradeMixin, ValueEstimate
+from modelbench.uid import HasUid
 
 
-class HazardDefinition(ABC):
+class HazardDefinition(ABC, HasUid):
+
+    def __init__(self):
```

Comment:
Given the structural similarities between hazards and benchmarks, would it make sense to be somewhat consistent in their class definitions? E.g. Make `_tests` a class variable which is set using a `tests` parameter in `__init__`?   Or conform `BenchmarkDefinition` to the current `HazardDefinition`.

Reply:
It would, but the way I'd make them parallel is by making it so that you don't need to know a Secret just to instantiate a Test. We really only need the Secret if a) we're doing a run, and b) the run isn't already fully cached. So I think the secrets are perhaps better passed in not at instantiation, but as part of the run. But I didn't want to redo all that just for this, so I went for the minimum higher distortion from the issues lower down.

Reply:
That makes sense!